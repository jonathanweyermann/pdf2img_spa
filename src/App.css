.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
}

.App-header {
  background-color: #282c34;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 1vmin);
  color: white;
}

.App-link {
  color: #09d3ac;
}

.move-up {
  margin-top: -25px;
}

.block {
  display: inline-block;
}

 .descriptor {
  color: #fff;
  font-size: 9px;
}

.navbar {
  background-color: #2C333A;
  color: #fff;
}

.navbar-brand {
  color: #FFF !important;
}

.input-padding {
  margin-bottom: 10px;
  margin-top: 10px;
  width: 250px;
}

.convert-button-padding {
  margin-left: 15px;
  padding: 9px !important;
  margin-bottom: 10px;
  margin-top: 10px;
}

.mt2 {
  margin-top: 20px;
}

.big {
  font-size: 24px;
}

.tight-container {
  max-width: 700px;
  color: #000000;
  text-align: left;
  display: inline-block;
  padding: 100px 0 0 0;
}

.center {
  text-align: center;
}

.error-message {
  background-color: green;
  padding: 20px;
}
