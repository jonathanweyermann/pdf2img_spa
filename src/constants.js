export let pagesToDisplay = 8;
export let imageBucket = process.env.REACT_APP_IMAGE_BUCKET;
