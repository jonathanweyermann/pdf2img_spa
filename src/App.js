import React, { Component } from 'react';
import './App.css';
import Main from './components/Main'
import About from './components/About'
import axios from 'axios'
import Amplify from 'aws-amplify';
import awsmobile from './aws-exports';
import { imageBucket } from './constants'
import FileStatus from './util/FileStatus'
import { Navbar, NavDropdown, Nav } from 'react-bootstrap';
import { BrowserRouter, Switch, Route } from "react-router-dom";

import { library } from '@fortawesome/fontawesome-svg-core'
import { faDownload } from '@fortawesome/free-solid-svg-icons'
import { createBrowserHistory } from 'history';
library.add(faDownload)

Amplify.configure(awsmobile);

axios.defaults.withCredentials = true

const history = createBrowserHistory();

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      previous_uploads: this.grabPreviousUploads()
    }

    this._isMounted = false;
    this.grabPreviousUploads = this.grabPreviousUploads.bind(this);
  }

  grabPreviousUploads = () => {
    if (localStorage.getItem('previous_uploads')) {
      var prev_uploads = JSON.parse(localStorage.getItem('previous_uploads'))
      var valid_uploads = prev_uploads.filter(this.checkPdfFileExistance);
      localStorage.setItem('previous_uploads', JSON.stringify(valid_uploads))
      return JSON.parse(localStorage.getItem('previous_uploads'));
    }
    else {
      return "[]"
    }
  }

  checkPdfFileExistance = (upload) => {
    return (FileStatus(`${imageBucket}pdfs/${upload.s3SafeFileName}`)===200)
  }

 header = () => {
   return (
     <Navbar className="navbar-dark" expand="sm" >
       <Navbar.Brand href="#home">PDF 2 Image<div className='descriptor'>Convert a PDF file to a set of optimized JPG images</div></Navbar.Brand>
       <Navbar.Toggle aria-controls="basic-navbar-nav" />
       <Navbar.Collapse id="basic-navbar-nav">
         <Nav className="mr-auto">
           <Nav.Link href="/">Home</Nav.Link>
           <NavDropdown title="Previous PDFs" id="basic-nav-dropdown">
              {
                this.state.previous_uploads.map((item, i) => {
                  return (<NavDropdown.Item key={i} href={`/uploaded/${i}`} state={ {item: item}}>{item["uploadFileName"]}</NavDropdown.Item>)
              })}
           </NavDropdown>
           <Nav.Link href="/about">About</Nav.Link>
         </Nav>
       </Navbar.Collapse>
     </Navbar>
    )
  }

  previous_uploads_update = (index) => {
    this.setState({previous_uploads: this.grabPreviousUploads()})
    if (index === -1) { index = this.state.previous_uploads.length-1 };
    history.push(`/uploaded/${index}`);
  }


  render() {

    return (
      <BrowserRouter history={history}>
        <div className="App">
          <header className="App-header">
            { this.header() }
          </header>
          <div>
            <Switch>
              <Route path="/about" component={About} />
              <Route path="/uploaded/:index" component={Main}>
                <Main previous_uploads={this.state.previous_uploads} previous_uploads_update={this.previous_uploads_update} />
              </Route>
              <Route path="/" component={Main}>
                <Main previous_uploads={this.state.previous_uploads} previous_uploads_update={this.previous_uploads_update} />
              </Route>
            </Switch>
          </div>
        </div>
      </BrowserRouter>
    );
  }
}

export default App;
