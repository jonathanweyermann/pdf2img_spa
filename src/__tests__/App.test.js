
import { expect as chai_expect } from 'chai';
import renderer from 'react-test-renderer';
let valid_uploads = '[{"uploadFileName":"template","numPages":2,"s3SafeFileName":"78559523template.pdf"},{"uploadFileName":"reactnative","numPages":242,"s3SafeFileName":"78559523reactnative.pdf"},{"uploadFileName":"TrainTickets","numPages":2,"s3SafeFileName":"78559523TrainTickets.pdf"}]';
var sinon = require('sinon');
import { imageBucket } from '../constants'

import React from 'react';
import { shallow, mount } from 'enzyme';
import App from '../App';

import mock from 'xhr-mock';

describe('First React component test with Enzyme', () => {
  it('renders welcome message', () => {
    const wrapper = mount(<App />);
    const welcome = <div className="center heading-font">Convert a PDF file to a set of optimized JPG images</div>
    expect(wrapper.contains(welcome)).toBe(true);
  });

  it('renders without crashing', () => {
    shallow(<App />);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<App />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders an `.btn-success', () => {
    const wrapper = mount(<App />);
    chai_expect(wrapper.find('.btn-success')).to.have.lengthOf(1);
  });

  it('there are no pdf dropdown items', () => {
    const wrapper = mount(<App />);
    chai_expect(wrapper.find('a.dropdown-item')).to.have.lengthOf(0);
  });
});
var xhr = null;
var requests = [];
describe('Previous uploads exist locally stored', () => {
  beforeEach(() => {
    localStorage.setItem('previous_uploads', valid_uploads)
    const xhrMockClass = () => ({
      open            : jest.fn(),
      send            : jest.fn(),
      setRequestHeader: jest.fn(),
      status: 200
    })
    window.XMLHttpRequest = jest.fn().mockImplementation(xhrMockClass)
  })

  it('creates a screenshot for the pdf dropdown view for previously uploaded pdfs', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('creates a pdf dropdown view for three previously uploaded pdfs', () => {
    const wrapper = mount(<App />);
    chai_expect(wrapper.find('a.dropdown-item')).to.have.lengthOf(3);
  });
});
