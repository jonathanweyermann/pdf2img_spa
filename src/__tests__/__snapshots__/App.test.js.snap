// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First React component test with Enzyme renders correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <nav
      className="navbar-dark navbar navbar-expand-sm navbar-light"
    >
      <a
        className="navbar-brand"
        href="#home"
      >
        PDF 2 Image
      </a>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          className="mr-auto navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="nav-link"
            data-rb-event-key="/"
            href="/"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Home
          </a>
          <div
            className="dropdown nav-item"
            onKeyDown={[Function]}
          >
            <a
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle nav-link"
              data-rb-event-key={null}
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
            >
              Previous PDFs
            </a>
            <div
              aria-labelledby={null}
              className="dropdown-menu"
            />
          </div>
          <a
            className="nav-link"
            data-rb-event-key="/about"
            href="/about"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            About
          </a>
        </div>
      </div>
    </nav>
  </header>
  <div>
    <div>
      <div
        className="main-style"
      >
        <div
          className="container"
        >
          <div
            className="tight-container"
          >
            <div
              className="center heading-font"
            >
              Convert a PDF file to a set of optimized JPG images
            </div>
            <div
              className="mt2"
            >
              This free tool converts PDFs to JPGs. It shows thumbnails of each page so you can easily enlarge and download only the files you wish to download
            </div>
            <div
              className="mt2"
            >
              <span
                className="big"
              >
                1
              </span>
               Click the 'Choose File' button and select a PDF file you wish to convert. Click 'Convert to JPG images' and wait for the conversion process to finish.
            </div>
            <div>
              <span
                className="big"
              >
                2
              </span>
               Download the results either file by file or click the DOWNLOAD ALL button to get them all at once in a ZIP archive.
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="container"
    >
      <input
        accept=".pdf"
        className="btn btn-success input-padding"
        onChange={[Function]}
        type="file"
        value=""
      />
      <button
        className="convert-button-padding btn btn-primary"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        Convert to JPG Images
      </button>
      <div>
        <img
          src="/upload.gif"
          style={
            Object {
              "marginTop": "40px",
              "maxWidth": "740px",
              "width": "100%",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Previous uploads exist locally stored creates a screenshot for the pdf dropdown view for previously uploaded pdfs 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <nav
      className="navbar-dark navbar navbar-expand-sm navbar-light"
    >
      <a
        className="navbar-brand"
        href="#home"
      >
        PDF 2 Image
      </a>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          className="mr-auto navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="nav-link"
            data-rb-event-key="/"
            href="/"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Home
          </a>
          <div
            className="dropdown nav-item"
            onKeyDown={[Function]}
          >
            <a
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle nav-link"
              data-rb-event-key={null}
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
            >
              Previous PDFs
            </a>
            <div
              aria-labelledby={null}
              className="dropdown-menu"
            >
              <a
                className="dropdown-item"
                href="/uploaded/0"
                onClick={[Function]}
                onKeyDown={[Function]}
                state={
                  Object {
                    "item": Object {
                      "numPages": 2,
                      "s3SafeFileName": "78559523template.pdf",
                      "uploadFileName": "template",
                    },
                  }
                }
              >
                template
              </a>
              <a
                className="dropdown-item"
                href="/uploaded/1"
                onClick={[Function]}
                onKeyDown={[Function]}
                state={
                  Object {
                    "item": Object {
                      "numPages": 242,
                      "s3SafeFileName": "78559523reactnative.pdf",
                      "uploadFileName": "reactnative",
                    },
                  }
                }
              >
                reactnative
              </a>
              <a
                className="dropdown-item"
                href="/uploaded/2"
                onClick={[Function]}
                onKeyDown={[Function]}
                state={
                  Object {
                    "item": Object {
                      "numPages": 2,
                      "s3SafeFileName": "78559523TrainTickets.pdf",
                      "uploadFileName": "TrainTickets",
                    },
                  }
                }
              >
                TrainTickets
              </a>
            </div>
          </div>
          <a
            className="nav-link"
            data-rb-event-key="/about"
            href="/about"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            About
          </a>
        </div>
      </div>
    </nav>
  </header>
  <div>
    <div>
      <div
        className="main-style"
      >
        <div
          className="container"
        >
          <div
            className="tight-container"
          >
            <div
              className="center heading-font"
            >
              Convert a PDF file to a set of optimized JPG images
            </div>
            <div
              className="mt2"
            >
              This free tool converts PDFs to JPGs. It shows thumbnails of each page so you can easily enlarge and download only the files you wish to download
            </div>
            <div
              className="mt2"
            >
              <span
                className="big"
              >
                1
              </span>
               Click the 'Choose File' button and select a PDF file you wish to convert. Click 'Convert to JPG images' and wait for the conversion process to finish.
            </div>
            <div>
              <span
                className="big"
              >
                2
              </span>
               Download the results either file by file or click the DOWNLOAD ALL button to get them all at once in a ZIP archive.
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="container"
    >
      <input
        accept=".pdf"
        className="btn btn-success input-padding"
        onChange={[Function]}
        type="file"
        value=""
      />
      <button
        className="convert-button-padding btn btn-primary"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        Convert to JPG Images
      </button>
      <div>
        <img
          src="/upload.gif"
          style={
            Object {
              "marginTop": "40px",
              "maxWidth": "740px",
              "width": "100%",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
